spec/pages_spec.rb

```diff
@@ -13,13 +64,28 @@
       visit p
     end
 
-    it 'has only valid internal hyperlinks' do
-      page.all(:css, 'a').each do |link|
-        next if link.text == '' || link[:href].match(/(http|\/\/).*/)
-        page.find(:xpath, link.path).click
-        expect(page.status_code).to be(200), "expected link '#{link.text}' to work"
-        visit p
+    it 'has valid internal hyperlinks' do
+      links = Links.new(page.all(:css, 'a'),urls)
+      links.each do |link|
+        url=link[:href]
+        if !is_uri? url
+          page.find(:xpath, link.path).click
+          expect(page.status_code).to be(200), "expected link '#{link.text}' to work"
+        end
       end
+      visit p
+    end
+
+    it 'has valid external hyperlinks' do
+      links = Links.new(page.all(:css, 'a'),urls)
+      links.each do |link|
+        url=link[:href]
+        if is_uri? url
+          result = Links.url_exists? url
```

Comment:
Thinking of it a bit more the method could return two values instead (you're not passing it anyway and using a global variable).